net.go

```
@@ -70,10 +70,31 @@ func doAPICall(endpoint Endpoint, req Request) (result []byte, err error) {
 	return doGet(uri)
 }
 
+func GetApiPath(publication, endpoint string, params map[string]string) (uri string) {
+	dp := domains[publication]
+	if "" == dp {
+		log.Printf("Pub requested: %s", publication)
+		return
+	}
+
+	uri = strings.Replace(urlTemplate, "{protocol}", protocol, 1)
+	uri = strings.Replace(uri, "{domainPort}", dp, 1)
+	uri = strings.Replace(uri, "{endpoint}", string(endpoint), 1)
+
+	p := url.Values{}
+	for key, value := range params {
+		p.Set(key, value)
+	}
+
+	uri += "?" + p.Encode()
+
+	return
+}
+
 func prepareAPIUri(endpoint Endpoint, req Request) (uri string) {
```

should also return an error if no domain for the requested publication